src/js.rs

```diff
@@ -36,12 +36,19 @@ pub(crate) fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error>
 
         match source {
             RngSource::Node(n) => {
-                // XXX(perf): `random_fill_sync` requires a `&mut [u8]` so we
-                // have to ensure the memory in `dest` is initialized.
-                let dest = uninit_slice_fill_zero(dest);
-
                 for chunk in dest.chunks_mut(NODE_MAX_BUFFER_SIZE) {
-                    if n.random_fill_sync(chunk).is_err() {
+                    // SAFETY: chunk is never used directly, the memory is only
+                    // modified via buf. Buf is passed directly to JavaScript
+                    // and crypto.randomFillSync does not resize the buffer. We
+                    // know the length is less than u32::MAX because of the
+                    // chunking above.
+                    // Note that this uses the fact that JavaScript doesn't
+                    // have a notion of "uninitialized memory", this is purely
+                    // a Rust/C/C++ concept.
+                    let buf = unsafe {
+                        Uint8Array::view_mut_raw(chunk.as_mut_ptr() as *mut u8, chunk.len())
+                    };
+                    if n.random_fill_sync(buf).is_err() {
```

Comment:
Shouldn't you extend the `unsafe` block to include `random_fill_sync`? I don't it's quite legal to have `buf` around in safe code since using it we can look into uninitialized memory. I mean something like this:
```rust
let res = unsafe {
    let buf = Uint8Array::view_mut_raw(chunk.as_mut_ptr() as *mut u8, chunk.len());
    n.random_fill_sync(buf)
};
if res.is_err() { .. }

Reply:
Makes sense to me. Do you think that:
```rust
let res = n.random_fill_sync(unsafe {
    Uint8Array::view_mut_raw(chunk.as_mut_ptr() as *mut u8, chunk.len())
});
```
would be ok?

Reply:
Yeah, it should be fine.